airflow/decorators/__init__.pyi

```diff
@@ -124,20 +124,14 @@ class TaskDecoratorCollection:
     @overload
     def virtualenv(self, python_callable: Function) -> Task[Function]: ...
     @overload
-    def branch(
-        self, python_callable: Optional[Callable] = None, multiple_outputs: Optional[bool] = None, **kwargs
-    ) -> TaskDecorator:
-        """Wraps a python function into a BranchPythonOperator
+    def branch(self, *, multiple_outputs: Optional[bool] = None, **kwargs) -> TaskDecorator:
```

Comment:
looks like you dropped an arg `python_callable`

also, seems the underlying method isn't kwargs only?  but maybe we just want to not encourage use of args

Reply:
The real implementation is intended to be used with two variants:

```python
@task.branch
def func(**context):
    ...

@task.branch(**kwargs)
def func(**context):
    ...
```

The first variant accepts exactly one positional argument `python_callable`. The second should not accept `python_callable`, but all the other arguments (normally as keywords, although theoratically you could do `@task.branch(None, True, ...)`). This type hint better reflects the intention, not the actual implementation details.

Reply:
ah makes sense